### ISSUE TITLE
Warning: Failed to get dockerfile path for image someorg/someimage:somehash | Error: docker images pull failed with: undefined

### ISSUE THREAD

**Issue by rabbagliettiandrea** [Azure/k8s-deploy#164]
### Repro steps:
- use a docker image that is built with Dockerfile not in the current working dir in the buildphase:
![image](https://user-images.githubusercontent.com/579861/147229485-2257258f-79ba-4e82-8731-1b94fb930671.png)

- deploy it normally with azure/k8s-deploy@v1.5 - 1.3 (**v1 is not affected**)
![image](https://user-images.githubusercontent.com/579861/147229630-228672bf-d7a1-4eda-9071-988a4f2987a2.png)

### Current behaviour:
![Senzanome](https://user-images.githubusercontent.com/579861/147230004-dedd4ef5-dc72-4544-966f-955c7d35e8ea.png)

### Expected behaviour:
no warning

Thank you so much guys for your work, keep it up! :)

**Comment by PadraigGalvin** [comment]
I'm getting a similar warning with k8s-deploy@v3.0. The cluster is running k8s v1.22.2-2.

```console
Failed to get dockerfile path for image ghcr.io/****/stats-service:c38dafa06ad75fc272c09ab5d57baeac702c033a: Error: The process '/usr/bin/docker' failed with exit code 1
```
The logs provide some additional info:

```console
Annotating resources
/usr/bin/docker pull ghcr.io/****/stats-service:c38dafa06ad75fc272c09ab5d57baeac702c033a
Error response from daemon: Head "https://ghcr.io/v2/****/stats-service/manifests/c38dafa06ad75fc272c09ab5d57baeac702c033a": unauthorized
Warning: Failed to get dockerfile path for image ghcr.io/****/stats-service:c38dafa06ad75fc272c09ab5d57baeac702c033a: Error: The process '/usr/bin/docker' failed with exit code 1 
```

My guess is that the action tries to pull the image locally to get information for updating annotations in the cluster. It seems that the GitHub Container Registry cannot be accessed in this way with the automatically provided token. A personal access token may be required instead.

**Comment by roehrijn** [comment]
Dealing with same issue here and I would highly appreciate to have a flag which controls the behaviour of pulling images. In most cases, this consumes a lot of resources in terms of time, traffic, storage, etc. ... only for having the Dockerfile's path in the annotation of the K8s resources. 

Furthermore I'm executing the action on a self-hosted runner based on RHEL, which has `podman` instead of `docker`.

**Comment by OliverMKing** [comment]
Thanks to @roehrijn for adding the ability to toggle this feature. This was released with version 3.1.

**Comment by nicmr** [comment]
Linking the PR that added the mentioned feature:
* #178 

Thank you for your work on this!
